internal/jsonutil/graphql.go

```diff
@@ -240,6 +284,38 @@ func (d *decoder) decode() error {
 	return nil
 }
 
+func copyTemplate(template reflect.Value) reflect.Value {
+	if isOrderedMap(template) {
+		// copy slice if it's actually an ordered map
+		return copyOrderedMap(template)
+	}
+	if template.Kind() == reflect.Map {
+		msg := fmt.Sprintf("unsupported template type `%v`, use [][2]interface{} for ordered map instead", template.Type())
+		panic(msg)
```

Comment:
should we return error, and let users decide whether to panic themselves?

Reply:
sure